### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by Ghabry**
Is a bit simpler then the Player CMake :+1:

Lacks that "Build custom liblcf" feature that Player cmake has... Will add this in another PR when I think this is worth it...

Had to rename "QGraphicsPaleteScene.h" to "qgraphicspaletescene.h" because the cpp is lowercase and this generates "vtable of QGraphicsPaleteScene undefined" errors.
In general the casing in the tools folder is so broken, this needs some work... (as the rest of the editor ;))

### CODE REVIEW

**Comment by Ghabry** [comment]
Hm, will replace that aux with a file list, lets try to be less lazy this time ;)

**Comment by carstene1ns** [comment]
This looks really good to me, compiles as well. Only have two minor things...

File: README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# EasyRPG Editor
+
+EasyRPG Editor is a game editor to create EasyRPG games. It can import
+RPG Maker 2000 and 2003 games using liblcf to read RPG Maker game data.
+
+EasyRPG Editor is part of the EasyRPG Project. More information is
+available at the project website: https://easyrpg.org/
+
+  https://easy-rpg.org/
```
Comment (carstene1ns) [inline README.md]: project_url

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,121 @@
+cmake_minimum_required(VERSION 3.1)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED on)
+
+project(EasyRPG_Editor CXX C)
```
Comment (carstene1ns) [inline CMakeLists.txt]: `C` is not needed here and cmake can skip some preparations then.

**Comment by Ghabry** [comment]
rebased and improvements added

**Pull request merged**